### ISSUE TITLE
can: "bosch,m-can-base": Warning "missing or empty reg/ranges property"

### ISSUE THREAD

**Issue by erwango** [zephyrproject-rtos/zephyr#42030]
**Describe the bug**
When building (any application) one of the boards that include can node with compatible "bosch,m-can-base",
the following warning is generated:
```
~/zephyrproject/zephyr/samples/synchronization/build/nucleo_h723zg/zephyr/zephyr.dts:2708.7-2734.5: Warning (simple_bus_reg): /soc/can: missing or empty reg/ranges property
```

**To Reproduce**
`west build -b nucleo_h723zg samples/hello_world`

**Expected behavior**
No warning generated

**Impact**
Annoyance

**Comment by henrikbrixandersen** [comment]
Agreed. We need to look into reworking the current hierarchy of Bosch M_CAN devicetree bindings (e.g. for STM32, we have `compatible = "bosch,m-can-base"` nodes with `compatible = "st,stm32-fdcan"` children, which is kind of weird).

**Comment by henrikbrixandersen** [comment]
@erwango First step in solving this: #45014

**Comment by henrikbrixandersen** [comment]
@erwango Second step in solving this: #44951

More to follow.

**Comment by str4t0m** [comment]
@henrikbrixandersen have you planned any further changes here?

I have drafted a PR to move STM32FD clock configuration from Kconfig to dts.
It would add the config to the instance node(can1, can2), although this is a property shared between can instances.
But the parent node is generic and would need bigger refactoring to be able to accommodate the very specific STM32 clock setup. [WIP](https://github.com/str4t0m/zephyr/tree/can_stm32fd_clock_control)

Therefore it would be interesting to hear about any progress or plans from your side, before I open a PR or rework this.

**Comment by henrikbrixandersen** [comment]
> @henrikbrixandersen have you planned any further changes here?

Yes! :)

> I have drafted a PR to move STM32FD clock configuration from Kconfig to dts. It would add the config to the instance node(can1, can2), although this is a property shared between can instances. But the parent node is generic and would need bigger refactoring to be able to accommodate the very specific STM32 clock setup. [WIP](https://github.com/str4t0m/zephyr/tree/can_stm32fd_clock_control)

Putting the clocks property within each child node goes hand-in-hand with my planned changes. I want us to get rid of the parent node altogether and move towards the Linux kernel devicetree bindings for the Bosch M_CAN IP.

> Therefore it would be interesting to hear about any progress or plans from your side, before I open a PR or rework this.

I started the work on transitioning the bindings a while back, but never got around to finishing it. I plan to pick it up again for it to be included in Zephyr v3.2. The major change is that the current way of determining the memory layout needs to be changed from being static and shared among all instances to being a per-instance base address and offsets in order to work with the individual dts nodes/new bindings.

**Comment by str4t0m** [comment]
> Putting the clocks property within each child node goes hand-in-hand with my planned changes. I want us to get rid of the parent node altogether and move towards the Linux kernel devicetree bindings for the Bosch M_CAN IP.

Great, I have now opened #49884 to move clock configuration from Kconfig to dts.

**Comment by mbolivar-nordic** [comment]
> I plan to pick it up again for it to be included in Zephyr v3.2.

@henrikbrixandersen is this still the plan?
